### ISSUE TITLE
[namespace] binary detection is not all that useful

### ISSUE THREAD

**Issue by xi** [xi/xi-keyring#3]
Examples:

- `d-feet` is a dbus UI implemented in python. The detected binary is `/usr/bin/python3.11`
- `keyring` is a CLI tool to interact with different keyrings. The detected binary is also `/usr/bin/python3.11`
- With the current approach to namespacing, both applications can see each other's secets
- If the system python is update to a new version, both applications lose access to their secrets

**Comment by xi** [comment]
I just realized that apparmor must have the same issue. I cannot find much discussion of this though.

**Comment by xi** [comment]
I was made aware of yet another issue with this approach: Applications can fake `/proc/{pid}/exe` by using LD_PRELOAD. For details, see section 4.1 in [this 1password security assessment](https://bucket.agilebits.com/security/SecfaultSecurity_Report_B5x_Security_Assessment_v1.1.pdf).

**Comment by xi** [comment]
Fixed in d0e372a3f12319b9ac1379f79fb295503bdffa91: I used `/proc/{pid}/root/.flatpak-info` instead. This is safe for flatpaks because the calling process is actually xdg-dbus-proxy, not the application itself. It is also the same approach that is used by [xdg-desktop-portal](https://github.com/flatpak/xdg-desktop-portal). The downside is that it only works for flatpaks. I am working on supporting the same mechanism for native packages in [xiwrap](https://github.com/xi/xiwrap).
